### DISCUSSION TITLE
Netbox Document Storage Plugin

### DISCUSSION THREAD

**Discussion by jasonyates** [netbox-community/netbox#9059] [General]
Hi,

I created a new 3.2 plugin (based off the great tutorial) that allows the management of Site, Device & Circuit specific documentation i.e. Floor plans, Network Diagrams, Contract documentation etc.

It's my first plugin, so i'd appreciate any feedback/features etc. You can see it [here](https://github.com/jasonyates/netbox-documents) and install netbox-documents from PyPi

Thanks

**Comment by empusas** [comment]
I like the idea. Maybe a versioning function would be beneficial. We use to keep the history of our diagrams to see what was changed when.
And a diagram per tenant. 
Another idea would be an embedded viewer. I know for example that not everybody might have Visio installed or people might use a Mac. If you have a MS sharepoint available this could be handy: [embeded VisioViewer](https://bvisual.net/2020/02/03/embedding-visio-documents-for-public-viewing/)

**Reply by candlerb** [reply]
[diagrams.net](https://www.diagrams.net/) (formerly draw.io) is cool. It can export as SVG/PNG/PDF etc files which can optionally include the source XML, so they're editable.

**Reply by empusas** [reply]
Yes, draw.io is cool. There is a feature to export a drawing by command line (headless). But there is an even better feature in Draw.io, to draw images from CSV. So we could create the image from data in netbox by creating an CSV first, then render a diagram. Only drawback is that this feature is not yet available on the command line.
[Drawings from CSV
](https://drawio-app.com/automatically-create-draw-io-diagrams-from-csv-files/)

**Comment by kopacko** [comment]
Can you tie a single document to multiple devices?

**Reply by adfrost81** [reply]
It would be also nice to be able to tie a single to document to multiple objects (like Sites or Site Groups)

**Comment by adfrost81** [comment]
We're using this very handy plugin for our installation; thanks for making it! I'd love to see this extended to other objects like Site Groups if possible.

It would also be nice if there were a way to upload via drag-and-drop, though this seems like a limitation of NetBox in general (unfortunately)
